x/staking/src/keeper/unbonding.rs

```diff
@@ -173,63 +188,69 @@ impl<
         if let Some(ref hooks) = self.hooks_keeper {
             hooks.after_validator_begin_unbonding(
                 ctx,
-                validator.get_cons_addr(),
+                validator.cons_addr(),
                 validator.operator_address.clone(),
             );
         }
         Ok(())
     }
 
-    pub fn unbond_all_mature_validators<DB: Database, CTX: TransactionalContext<DB, SK>>(
-        &self,
-        ctx: &mut CTX,
-    ) -> anyhow::Result<()> {
-        // TODO: time in ctx
-        let block_time = Utc::now();
+    pub fn unbond_all_mature_validators<DB: Database>(&self, ctx: &mut BlockContext<'_, DB, SK>) {
+        let block_time = ctx
+            .header()
+            .expect("BlockContext always has header")
+            .time
+            .as_ref()
+            .expect("Expected timestamp in block context header.");
```

Comment:
As above. Probably replace with unwrap and TODOs

Reply:
added